scripts/cl_acquisition.py

```diff
@@ -60,120 +66,268 @@ def __init__(self):
+        self._last_sem_data = []  # DataArray of the last spot acquisition
+        self._sem_data_received = threading.Event()
 
-        self.acq_left = 0 # how many points left to scan
-        self.acq_spect_buf = [] # list that will receive the acquisition data (numpy arrays of shape (N,1))
-        self.acq_complete = threading.Event()
+    def get_optical_image(self) -> DataArray:
+        """
+        Acquires an image from the CCD/spectrometer
+        """
+        return self.spect.data.get()
+
+    def get_spot_positions(self) -> ndarray:
+        """
+        Compute the positions of the e-beam for each point in the RoI.
+        Note that the points correspond to the *center* of squares dividing the
+        RoI. So the first and last points are *not* at the border of the RoI,
+        but a little bit away from the border.
+        :return: (ndarray of floats of shape (Y,X,2)): each value is for a
+          given Y,X in the rep grid -> 2 floats corresponding to the
+          translation X,Y. Note that the dimension order is different between
+          index and content, because X should be scanned first, so it's last
+          dimension in the index.
+        """
+        rep = self.repetition
+        roi = self.roi
+        width = (roi[2] - roi[0], roi[3] - roi[1])
 
-    def set_roi(self, rect):
-        # save the resolution
-        res = self.escan.resolution.value
-        center = ((rect[0] + rect[2]) / 2, (rect[1] + rect[3]) / 2)
-        width = (rect[2] - rect[0], rect[3] - rect[1])
+        # Take into account the "border" around each pixel
+        pxs = (width[0] / rep[0], width[1] / rep[1])
+        lim = (roi[0] + pxs[0] / 2, roi[1] + pxs[1] / 2,
+               roi[2] - pxs[0] / 2, roi[3] - pxs[1] / 2)
 
         shape = self.escan.shape
-        scale = (1 / width[0], 1 / width[1])
-        # translation is distance from center (situated at 0.5, 0.5), can be floats
-        trans = (shape[0] * (center[0] - 0.5), shape[1] * (center[1] - 0.5))
-        # always in this order
-        self.escan.scale.value = scale # might update the resolution & translation
-        self.escan.resolution.value = res
-        self.escan.translation.value = trans
+        # convert into SEM translation coordinates: distance in px from center
+        # (situated at 0.5, 0.5), can be floats
+        lim_main = (shape[0] * (lim[0] - 0.5), shape[1] * (lim[1] - 0.5),
+                    shape[0] * (lim[2] - 0.5), shape[1] * (lim[3] - 0.5))
+        logging.info("Generating points in the SEM area %s, from rep %s and roi %s",
+                     lim_main, rep, roi)
 
-    def acquire_cube(self):
+        pos = numpy.empty((rep[1], rep[0], 2), dtype=numpy.float)
+        posy = pos[:,:, 1].swapaxes(0, 1)  # just a view to have Y as last dim
+        posy[:,:] = numpy.linspace(lim_main[1], lim_main[3], rep[1])
+        # fill the X dimension
+        pos[:,:, 0] = numpy.linspace(lim_main[0], lim_main[2], rep[0])
```

Comment:
PEP8 recommendations regarding whitespace after comma.
```suggestion
        posy = pos[:, :, 1].swapaxes(0, 1)  # just a view to have Y as last dim
        posy[:, :] = numpy.linspace(lim_main[1], lim_main[3], rep[1])
        # fill the X dimension
        pos[:, :, 0] = numpy.linspace(lim_main[0], lim_main[2], rep[0])
```